conf/schemas_test.go

```diff
@@ -0,0 +1,137 @@
+package conf
+
+import (
+	. "github.com/smartystreets/goconvey/convey"
+	"regexp"
+	"testing"
+)
+
+func schemasForTest() Schemas {
+	return NewSchemas([]Schema{
+		{
+			Name:    "a",
+			Pattern: regexp.MustCompile("^a\\..*"),
+			Retentions: []Retention{
+				NewRetentionMT(10, 3600, 60*10, 0, true),
+				NewRetentionMT(3600, 86400, 60*60*6, 0, true),
+			},
+		},
+		{
+			Name:    "b",
+			Pattern: regexp.MustCompile("^b\\..*"),
+			Retentions: []Retention{
+				NewRetentionMT(1, 60, 60*10, 0, true),
+				NewRetentionMT(30, 120, 60*30, 0, true),
+				NewRetentionMT(600, 86400, 60*60*6, 0, true),
+			},
+		},
+		{
+			Name:    "default",
+			Pattern: regexp.MustCompile(".*"),
+			Retentions: []Retention{
+				NewRetentionMT(1, 60, 60*10, 0, true),
+				NewRetentionMT(60, 3600, 60*60*2, 0, true),
+				NewRetentionMT(600, 86400, 60*60*6, 0, true),
+				NewRetentionMT(3600, 86400*7, 60*60*6, 0, true),
+			},
+		},
+	})
+}
+
+func TestMatch(t *testing.T) {
+	schemas := schemasForTest()
+	Convey("When matching against first schema", t, func() {
+		Convey("When metric has 10s raw interval", func() {
+			id, schema := schemas.Match("a.foo", 10)
+			So(id, ShouldEqual, 0)
+			So(schema.Name, ShouldEqual, "a")
+			So(schema.Retentions[0].SecondsPerPoint, ShouldEqual, 10)
+		})
+		Convey("When metric has 30s raw interval", func() {
+			id, schema := schemas.Match("a.foo", 30)
+			So(id, ShouldEqual, 0)
+			So(schema.Name, ShouldEqual, "a")
+			So(schema.Retentions[0].SecondsPerPoint, ShouldEqual, 10)
+		})
+		Convey("When metric has 1s raw interval", func() {
```

Comment:
any particular reason behind the ordering 10s -> 30s -> 1s ? i would do them in ascending order.